rules/test.bzl

```diff
@@ -19,8 +19,81 @@ _IOS_TEST_KWARGS = [
     "shard_count",
     "flaky",
     "frameworks",
+    "test_filter",
 ]
 
+def _split_ios_test(name, test_rule, test_suite_rule, apple_library, split_name_to_kwargs, infoplists_by_build_setting = {}, **kwargs):
+    """
+    Builds and packages iOS Unit/UI Tests splitting the package into as many "splits" specified in the split_name_to_kwargs.
+
+    Args:
+        name: The name of the unit test.
+        test_rule: The underlying rules_apple test rule.
+        test_suite_rule: The underlying rules_apple test suite rule.
+        apple_library: The macro used to package sources into a library.
+        split_name_to_kwargs: A dictionary of suffixes to kwargs that will be passed into the "split" test bundle. The suffix will be appended to the name of the suite.
+        infoplists_by_build_setting: A dictionary of infoplists grouped by bazel build setting.
+
+                                     Each value is applied if the respective bazel build setting
+                                     is resolved during the analysis phase.
+
+                                     If '//conditions:default' is not set the value in 'infoplists'
+                                     is set as default.
+        **kwargs: Arguments passed to the apple_library and test_rule rules as appropriate.
+    """
+
+    runner = kwargs.pop("runner", kwargs.pop("runners", None))
+    ios_test_kwargs = {arg: kwargs.pop(arg) for arg in _IOS_TEST_KWARGS if arg in kwargs}
+    ios_test_kwargs["data"] = kwargs.pop("test_data", [])
+    if ios_test_kwargs.get("test_host", None) == True:
+        ios_test_kwargs["test_host"] = "@build_bazel_rules_ios//rules/test_host_app:iOS-%s-AppHost" % ios_test_kwargs.get("minimum_os_version")
+
+    # Deduplicate against the test deps
+    if ios_test_kwargs.get("test_host", None):
+        host_args = [ios_test_kwargs["test_host"]]
+    else:
+        host_args = []
+    library = apple_library(name = name, namespace_is_module_name = False, platforms = {"ios": ios_test_kwargs.get("minimum_os_version")}, testonly = True, **kwargs)
+
+    # Setup framework middlemen - need to process deps and libs
+    fw_name = name + ".framework_middleman"
+    framework_middleman(name = fw_name, framework_deps = kwargs.get("deps", []) + library.lib_names, testonly = True, tags = ["manual"])
+    frameworks = [fw_name] + ios_test_kwargs.pop("frameworks", [])
+
+    dep_name = name + ".dep_middleman"
+    dep_middleman(name = dep_name, deps = kwargs.get("deps", []) + library.lib_names, testonly = True, tags = ["manual"], test_deps = host_args)
+
+    tests = []
+    for suffix, split_kwargs in split_name_to_kwargs.items():
+        all_kwargs = {}
+        all_kwargs.update(ios_test_kwargs)
+        all_kwargs.update(split_kwargs)
+
+        if "runner" in kwargs and "runners" in kwargs:
```

Comment:
Any reason to not pass this via: `split_name_to_kwargs` - then we can simplify / move validation logic out of this rule to the place the `split` kwargs are combined